### ISSUE TITLE
Include more options for niche definitions

### ISSUE THREAD

**Issue by timtreis** [scverse/squidpy#789]
Currently, squidpy focusses mostly on local [spatial neighborhoods ](https://squidpy.readthedocs.io/en/stable/notebooks/tutorials/tutorial_vizgen_mouse_liver.html#neighborhood-enrichment) for niche analysis purposes. However, there are other definitions which one could use. Including other options could increase tool usage and enable other usecases, f.e. nische-detection-robustness testing?

Pinging @melonora @LLehner @MohammedZidane

**Comment by timtreis** [comment]
Hey @LLehner, let's discuss the feature in this issue. Here there's the monkeybread package that we mentioned during the meeting: https://monkeybread.readthedocs.io/en/latest/

**Comment by LLehner** [comment]
**Neighborhood based:**
- [monkeybread](https://monkeybread.readthedocs.io/en/latest/generated/monkeybread.calc.cellular_niches.html#monkeybread.calc.cellular_niches) (clustering of neighborhood profiles)
- [CellCharter](https://www.biorxiv.org/content/10.1101/2023.01.10.523386v2) (performs dimreduction, neighborhood calculation, aggregation and clustering)

**VI-based:**
- [DestVI](https://www.nature.com/articles/s41587-022-01272-8) ("Unlike other methods, DestVI learns both discrete cell-type-specific profiles and continuous sub-cell-type latent variations using a conditional deep generative model")

**OT-based:**
- [SPOT](https://openreview.net/forum?id=knxP0g5K8A) (Cluster cells based on their niche similarity, not their expression similarity)

**GRN based:**
- [NicheHotSpotter](https://doi.org/10.1111/febs.15832) ("NicheHotSpotter calculates the probability that a signalling node participates in niche–stem cell signalling from the stationary distribution of a finite discrete time-homogenous Markov chain model")

**Cell-Cell-Communiation based:**
One could reason about niches by analyzing cellular interactions:
- [CellPhoneDB](https://doi.org/10.1038/s41596-020-0292-x)
- [NicheNet](https://doi.org/10.1038/s41592-019-0667-5)
- [NCEM](https://doi.org/10.1038/s41587-022-01467-z)

**pre-trained (foundational) models:**
Perhaps one could utlize pre-trained (foundational) models to predict niches.
- [Nicheformer](https://www.biorxiv.org/content/biorxiv/early/2024/04/17/2024.04.15.589472.full.pdf)

**Comment by timtreis** [comment]
Amazing, thank you! Could you also briefly look into the respective requirements? Ideally I'd hope that there is a small subset that we can add without adding to our dependencies (`core`) and the other methods will complain about their specific solvers or whatever not being installed when they're called (`addon`). 

Then we could start with the `core` methods and also merge them without adding to the dependency load of the base package before things are evaluated and stable and later on cast the `addon` ones into the template we talked about

**Comment by chiarasch** [comment]
Great progress in #831! @migueLib and I think about adding Kasumi from @jtanevski to squidpy, also referring to the recent addition of sliding_windows in #859

**Comment by LLehner** [comment]
Hi @chiarasch, thank you for the offer! That would indeed be a great addition. I just skimmed through the paper and if i understand correctly the third level representation which is part of the output is the cluster label, which would be the niche a cell is assigned to, right? 

Another question: If you are set on implementing this, do you have a rough timeline in mind until when its ready? Just asking because if you plan to start right now you could add it to #831, otherwise make a new PR, as that one should get merged soon.

**Comment by jtanevski** [comment]
Hi @LLehner. You are right that the third level is the output. However, Kasumi outputs a class label per tissue patch/window and not per cell. That makes it a bit different from the related approaches. The package is updated and it allows also for clustering of windows based on the composition of cell types or marker abundance/expression in general as an alternative that, as a scope and representation, lends itself also to application of other approaches.

**Comment by LLehner** [comment]
> Hi @LLehner. You are right that the third level is the output. However, Kasumi outputs a class label per tissue patch/window and not per cell. 

Ah, i see. Well that shouldn't be a problem, i mean you can just assign the patch label to every cell that resides within this patch (if one is interested in labels per cell). But just to understand correctly the third level label can be interpreted as a niche label?

**Comment by jtanevski** [comment]
> > Hi @LLehner. You are right that the third level is the output. However, Kasumi outputs a class label per tissue patch/window and not per cell.
> 
> Ah, i see. Well that shouldn't be a problem, i mean you can just assign the patch label to every cell that resides within this patch (if one is interested in labels per cell). But just to understand correctly, can the third level label be interpreted as a niche label?

In principle yes, but it's going to be tricky if you want to do that since the windows can overlap leading to a fuzzy instead of crisp assignment :)

**Comment by chiarasch** [comment]
Thanks for checking @LLehner! It is indeed a bit trickier to compare it to the more "classical" niche identification methods, also in the context of how you are comparing them atm. We thought about adding it during the hackathon in November, so I would create another PR and we can deepen our discussions then! :) 

Out of interest, which other methods do you plan to add to the current PR?

**Comment by timtreis** [comment]
I'll let @LLehner elaborate more on the exact method choices, but the core thought is "We don't want crazy imports or massive reimplementation work" 👌 

Just wanted to ping @FrancescaDr who's working on the sliding window PR. This method could f.e. be a cool way to demonstrate both functionalities later for `squidpy-notebooks`, wdyt?

**Comment by LLehner** [comment]
>We thought about adding it during the hackathon in November, so I would create another PR and we can deepen our discussions then! :)

Great, looking forward to it!

> Out of interest, which other methods do you plan to add to the current PR?

[SPOT](https://openreview.net/pdf?id=knxP0g5K8A) will be added next. Then we would also like to implement [BANKSY](https://www.nature.com/articles/s41588-024-01664-3) and perhaps a HMRF-based approach.

**Comment by grst** [comment]
Found a quite recent benchmark on biorxiv that compares different algorithms and includes more options than what's listed here: 
https://www.biorxiv.org/content/10.64898/2026.02.27.708202v1.full

**Comment by LLehner** [comment]
Thanks @grst, we will definitely update this in the future!

**Comment by grst** [comment]
Another, relatively new option: [scNiche](https://www.nature.com/articles/s41467-025-57029-9).
